packages/cdktf/lib/tfExpression.ts

```diff
@@ -374,6 +374,72 @@ export function call(name: string, args: Expression[]) {
   return new FunctionCall(name, args) as IResolvable;
 }
 
+export const FOR_EXPRESSION_KEY = ref("key");
+export const FOR_EXPRESSION_VALUE = ref("val");
+
+/**
+ * https://www.terraform.io/docs/language/expressions/for.html
+ */
+class ForExpression extends TFExpression {
+  constructor(
+    private input: Expression,
+    private valueExpression: Expression,
+    private keyExpression?: Expression
+  ) {
+    super({ input, valueExpression, keyExpression });
+  }
+
+  public resolve(context: IResolveContext): string {
+    const suppressBraces = context.suppressBraces;
+    context.suppressBraces = true;
+    const key = this.resolveArg(context, FOR_EXPRESSION_KEY);
+    const value = this.resolveArg(context, FOR_EXPRESSION_VALUE);
+    const input = this.resolveArg(context, this.input);
+    const valueExpr = this.resolveArg(context, this.valueExpression);
+    const keyExpr = this.keyExpression
```

Comment:
Nit: I'd inline this var to remove a layer of indirection

Reply:
Am I missing something or did you miss the next two lines? 😅